synths/core-synths.scd

```diff
@@ -116,9 +116,11 @@ live coding them requires that you have your SuperDirt instance in an environmen
 			//phase =  Sweep.ar(1, rate * BufSampleRate.ir(bufnum)) + (BufFrames.ir(bufnum) * begin);
 
 			numFrames = BufFrames.ir(bufnum);
-			// try windows that are 1/29th the width of the result, but not too small or large or it sounds weird
-			windowSize = numFrames * timescale / 29.0;
-			windowSize = windowSize.clip(1000*timescale, BufSampleRate.ir(bufnum)/10);
+			// Picking the right window size is a tricky thing, something around 2000 samples is usually
+			// OK for sounds of moderate length (around 0.5 to several seconds long).
+			// But this is also scaled below to be a bit smaller/larger based on desired playback rate.
+			// If it's still not good, you can use the "timescalewin" parameter to multiply the window size
+			windowSize = timescale.clip(0.1,2) * 0.05 * BufSampleRate.ir(bufnum) * timescalewin;
```

Comment:
`windowSize` and `timescale` can be varied completely independently and the algorithm still works, but may sound odd for certain combinations.  I'm really clipping on windowSize values, but a "good" windowSize tends to scale somewhat with the length of the effect.  I'm not sure what you mean about the sustain.

The `windowIndex` calculation is correct, but unnecessarily obscure.  I added a variable to clarify it a bit (and it might save a repeated computation).

Reply:
>I'm not sure what you mean about the sustain.

Let's say e.g. you are setting `# timescale 0.09` for a sample that is 1 second long. Then your playback synth will have a total duration (=sustain) of 0.09 seconds. But your window will be `(0.1 * 0.05)`. Or, when you set `# timescale 3`, your synth will be 3 seconds, but the window will be `(2 * 0.05)`. But also, in line 124 (`timescaleStep = windowSize / timescale;` e.g. for a very large timescale, the timescalestep will be very small, but the window won't get smaller because of the clipping.

I am not sure if all that is a problem, it just looks like a source of hard to trace errors. It might be easier to reason about when you limit the stretching before everything else?

Reply:
Sorry to have left this dangling. I meant that because `timescaleStep` is clipped, the overlaps will become larger when `timescale < 0.1`. Could you check if that is really correct? If that is what you want, it might be good to make a comment.

Reply:
I'll take a look!  I have since noticed that a combo of frequency shift (`fshift`) and speed modification works pretty well for smaller amounts of stretching.